src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Client.Internal.Requests;
+
+namespace Microsoft.Identity.Client.Cache
+{
+    internal static class SuggestedWebCacheKeyFactory
```

Comment:
I decided to centralize the calculation of the key even though we have 1 class for each request type. In the end, the design is cleaner.